PocketKit/Sources/PocketKit/PocketAppDelegate.swift

```diff
@@ -94,6 +94,7 @@ public class PocketAppDelegate: UIResponder, UIApplicationDelegate {
         do {
             let attempted = try legacyUserMigration.perform()
             if attempted {
+                Services.shared.braze.signedInUserDidBeginMigration()
```

Comment:
Does this run after or before the migration runs? We want this as soon as possible so we can trigger an In app message

Reply:
as far as I can tell it'll run after a successful migration (happens inside `perform()`) and before anything gets loaded, am I right @CMasterson ?

Reply:
ok @CMasterson we want this to happen after we determined they need a migration, but before it is run

Reply:
There is an `isRequired` property in `LegacyUserMigration` that might come in handy

Reply:
I think it's very slightly after, but we know that the migration was actually attempted.
I can dig down a little and see if there's a spot we know the migration will be tried.

Reply:
Modified to pass a closure into the `perform` function.
That closure is executed if a migration is about to be performed.